cloudinit/config/cc_puppet.py

```diff
@@ -132,7 +140,8 @@ def _autostart_puppet(log):
             capture=False,
         )
     elif subp.which("systemctl"):
-        subp.subp(["systemctl", "enable", "puppet.service"], capture=False)
+        service = f"{get_service_name()}.service"
+        subp.subp(["systemctl", "enable", service], capture=False)
```

Comment:
why don't we call `distro.service` here, that's what it's for

Reply:
I see this will require us to pass a cloud instance to the function. Can we have a separate PR for such and other refactors?

Reply:
fair.